src/languageTools/LanguageClient/ProtocolAdapter.js

```diff
@@ -0,0 +1,337 @@
+}
+
+function didChangeTextDocument(connection, params) {
+    connection.sendNotification(protocol.DidChangeTextDocumentNotification.type, params);
+}
+
+function didCloseTextDocument(connection, params) {
+    connection.sendNotification(protocol.DidCloseTextDocumentNotification.type, params);
+}
+
+function didSaveTextDocument(connection, params) {
+    connection.sendNotification(protocol.DidSaveTextDocumentNotification.type, params);
+}
+
+function didChangeWorkspaceFolders(connection, params) {
+    connection.sendNotification(protocol.DidChangeWorkspaceFoldersNotification.type, params);
+}
+
+/** For Request messages */
+function completion(connection, params) {
+    return connection.sendRequest(protocol.CompletionRequest.type, params);
+}
+
+function completionItemResolve(connection, params) {
+    return connection.sendRequest(protocol.CompletionResolveRequest.type, params);
+}
+
+function signatureHelp(connection, params) {
+    return connection.sendRequest(protocol.SignatureHelpRequest.type, params);
+}
+
+function gotoDefinition(connection, params) {
+    return connection.sendRequest(protocol.DefinitionRequest.type, params);
+}
+
+function gotoDeclaration(connection, params) {
+    return connection.sendRequest(protocol.DeclarationRequest.type, params);
+}
+
+function gotoImplementation(connection, params) {
+    return connection.sendRequest(protocol.ImplementationRequest.type, params);
+}
+
+function findReferences(connection, params) {
+    return connection.sendRequest(protocol.ReferencesRequest.type, params);
+}
+
+function documentSymbol(connection, params) {
+    return connection.sendRequest(protocol.DocumentSymbolRequest.type, params);
+}
+
+function workspaceSymbol(connection, params) {
+    return connection.sendRequest(protocol.WorkspaceSymbolRequest.type, params);
+}
+
+/**
+ * Server commands
+ */
+function initialize(connection, params) {
+    var _params = {
+        rootPath: params.rootPath,
+        rootUri: Utils.pathToUri(params.rootPath),
+        processId: process.pid,
+        capabilities: params.capabilities
+    };
+
+    return connection.sendRequest(protocol.InitializeRequest.type, _params);
+}
+
+function initialized(connection) {
+    connection.sendNotification(protocol.InitializedNotification.type);
+}
+
+function shutdown(connection) {
+    return connection.sendRequest(protocol.ShutdownRequest.type);
+}
+
+function exit(connection) {
+    connection.sendNotification(protocol.ExitNotification.type);
+}
+
+function processRequest(connection, message) {
+    return _constructParamsAndRelay(connection, message.type, message.params);
+}
+
+function processNotification(connection, message) {
+    _constructParamsAndRelay(connection, message.type, message.params);
+}
+
+function attachOnNotificationHandlers(connection, handler) {
+    function _callbackFactory(type) {
+        switch (type) {
+        case ToolingInfo.protocol.ShowMessageNotification.type:
```

Comment:
ToolingInfo Object does not have protocol properties?

Reply:
Yeah it doesn't.